crates/environ/src/component/types.rs

```diff
@@ -493,31 +497,37 @@ impl ComponentTypesBuilder {
     pub fn convert_type(
         &mut self,
         types: types::TypesRef<'_>,
-        id: types::TypeId,
+        id: types::ComponentAnyTypeId,
     ) -> Result<TypeDef> {
-        Ok(match &types[id] {
-            types::Type::Defined(_) => TypeDef::Interface(self.defined_type(types, id)?),
-            types::Type::Module(_) => TypeDef::Module(self.convert_module(types, id)?),
-            types::Type::Component(_) => TypeDef::Component(self.convert_component(types, id)?),
-            types::Type::ComponentInstance(_) => {
+        Ok(match id {
+            types::ComponentAnyTypeId::Defined(id) => {
+                TypeDef::Interface(self.defined_type(types, id)?)
+            }
+            // TODO
+            // types::ComponentAnyTypeId::Module(_) => {
+            //     TypeDef::Module(self.convert_module(types, id)?)
+            // }
```

Comment:
Never got back to this TODO?

Reply:
Oops yes, just needed to delete the code

Reply:
For the curious, this TODO led to https://github.com/WebAssembly/component-model/issues/269